docs/docs/tutorials/orca-yarn-tutorial.md

```diff
@@ -0,0 +1,274 @@
+
+```bash
+# For Ubuntu
+sudo apt-get install openjdk-8-jre
+export JAVA_HOME=/usr/lib/jvm/java-8-openjdk-amd64/
+
+# For CentOS
+su -c "yum install java-1.8.0-openjdk"
+export JAVA_HOME=/usr/lib/jvm/java-1.8.0-openjdk-1.8.0.282.b08-1.el7_9.x86_64/jre
+
+export PATH=$PATH:$JAVA_HOME/bin
+java -version  # Verify the version of JDK.
+```
+
+## Setup Spark Environment
+We need to set environment variables `${SPARK_HOME}` as follows:
+```bash
+export SPARK_HOME=the folder path where you extract the Spark package
+```
+
+## Setup Hadoop Environment
+Check the Hadoop setup and configurations of our cluster. Make sure we correctly set the environment variable HADOOP_CONF_DIR, which is needed to initialize Spark on YARN:
+
+```bash
+export HADOOP_CONF_DIR=the directory of the hadoop and yarn configurations
+```
+
+## Install Needed Python Libraries
+We need first to use conda to prepare the Python environment on the local machine where we submit our application. Create a conda environment, install BigDL and all the needed Python libraries in the created conda environment:
+``` bash
+conda create -n bigdl python=3.7  # "bigdl" is conda environment name, you can use any name you like.
+conda activate bigdl
+# Use conda or pip to install all the needed Python dependencies in the created conda environment.
+pip install bigdl
+pip install conda-pack # Use conda-pack to pack the conda environment to an archive
+```
+### Options
+* When you are running a program on Ray backend, please install Ray as below:
+```bash
+pip install ray[default]
+```
+* When you are running a TensorFlow2 model, please install TensorFlow as below:
+```bash
+pip install tensorflow==2.6.0 keras==2.6.0 # When you are running a tensorflow model.
+```
+
+## For CDH Users
+If your CDH cluster has already installed Spark, the CDH’s Spark might be conflict with the pyspark installed by pip required by BigDL. Thus before running BigDL applications, you should unset all the Spark related environment variables. You can use `env | grep SPARK` to find all the existing Spark environment variables.
+
+Also, a CDH cluster’s `HADOOP_CONF_DIR` should be `/etc/hadoop/conf` on CDH by default.
+
+# Quick Start
+In the following part, we will show you three ways that BigDL supports to submit and run our programs on Yarn:
+* Use a built-in function
+* Use a BigDL provided script
+* Use a spark-submit script
+
+# Run Programs with built-in function
+This is the easiest and most recommended way to run BigDL on YARN, we only need to prepare the environment on the driver machine, all dependencies would be automatically packaged and distributed to the whole Yarn cluster.
+
+## Yarn Client
+When running programs on yarn-client mode by built-in function, we need to set `--cluster_mode` argument as `yarn-client` or `yarn` to create an OrcaContext on yarn for yarn-client mode.
```

Comment:
Show code snippet for how to call `init_orca_context` in this case (as well other cases later)